### ISSUE TITLE
A Great Work

### ISSUE THREAD

**Issue by ghost** [KinWaiCheuk/demucs_lightning#6]
Thank you for your work. 
1. I trained the model you provided. I want to know whether can provide test code. I write one myself, but the effect is not very good. I only use 20 epoch at present, and I don't know whether epoch is enough .
2. when I test ，would I need to  normalize the test wav？
I hope you can help me with the question

**Comment by heungky** [comment]
Hi, thank you for using demucs_lightning. 

1. We added test.py for evaluation purpose. You can refer to [README.md](https://github.com/KinWaiCheuk/demucs_lightning#Testing-pretrained-model) for more information. By using test.py , you are able to evaluate the pretrained mode from an existing checkpoint. You are right that 20 epochs might not be enough, as the original demucs uses 360 epochs as the default setting.

2. By default we have normalization for train, valid, test set audio. You can refer to [config.yaml](https://github.com/KinWaiCheuk/demucs_lightning/blob/master/conf/config.yaml). If you are using your own custom test set, we recommend you to also normalize the audio.

Hope it helps :)

**Comment by qcj1206** [comment]
hello，I train the HDemus model by myself. However, I found that there is a hissing noise in the separated audio(vocals). I followed your steps completely, except that setting batch=2.Can you help me or give me a resume_checkpoint to check!

**Comment by heungky** [comment]
Hi, thanks for your feedback. 
Regarding the hissing noise in the vocals separated audio, may I know your Test/nsdr_vocals when batch_size =2 ? 

From our experiments, the Test/nsdr_vocals are in the following:
* Hdemucs_10epoch_batch=2 is 5.49
* Hdemucs_10epoch_batch=4 is 5.40

For your reference, you can get the checkpoint file of the experiments above via [here](https://sutdapac-my.sharepoint.com/:f:/g/personal/kwanyee_heung_sutd_edu_sg/EkmolXU_GwBErxNdT2ax5R4BLWun7MRRuF46fENm3WeczQ?e=iIFEmB)

We noticed some of the test/label/vocals audio that already came with hissing noise (e.g. test/Arise - Run Run Run). You might want to cross reference with the test label audio as well :)

**Comment by SEMLLYCAT** [comment]
Hello, I would like to ask that the results of each test or inference of the same model are inconsistent

**Comment by SEMLLYCAT** [comment]
> 



> 您好，感谢您的反馈。 关于人声分离音频中的嘶嘶声，我可以知道你的 Test/nsdr_vocals when batch_size =2 吗？
> 
> 根据我们的实验，Test/nsdr_vocals 位于以下位置：
> 
> * hdemucs_10epoch_batch=2 为 5.49
> * hdemucs_10epoch_batch=4 为 5.40
> 
> [供您参考，您可以通过此处](https://sutdapac-my.sharepoint.com/:f:/g/personal/kwanyee_heung_sutd_edu_sg/EkmolXU_GwBErxNdT2ax5R4BLWun7MRRuF46fENm3WeczQ?e=iIFEmB)获取上述实验的检查点文件[](https://sutdapac-my.sharepoint.com/:f:/g/personal/kwanyee_heung_sutd_edu_sg/EkmolXU_GwBErxNdT2ax5R4BLWun7MRRuF46fENm3WeczQ?e=iIFEmB)
> 
> 我们注意到一些测试/标签/人声音频已经带有嘶嘶声（例如测试/Arise - Run Run Run）。您可能还想与测试标签音频进行交叉引用 :)
Hell0! When I repeat training according to your project, checkpionts with lower loss is not better than before. Hello, please kindly ask for your convenience help me or give me a resume_checkpoint to check!

**Comment by SEMLLYCAT** [comment]
The link you provided cannot be opened

**Comment by SEMLLYCAT** [comment]
> 你好，我自己训练HDemus模型。但是，我发现分离的音频（人声）中有嘶嘶声。除了设置 batch=2 之外，我完全按照你的步骤操作。你能帮帮我吗？或者给我一个 resume_checkpoint 来检查！

Hello, is it convenient to offer checkpiont

**Comment by turian** [comment]
@SEMLLYCAT You can load htdemucs checkpoints into this repo. https://github.com/facebookresearch/demucs

**Comment by SEMLLYCAT** [comment]
Thank you for your reply. I may not have expressed myself clearly. I wonder if it is convenient for you to provide the pre-training model of your version of pytorch_lightning. I have been unable to reproduce the effect of the original paper (V3) through the default parameters you provided. I believe the version you provided is very useful. Thank you very much! Looking forward to your recovery!
















At 2023-02-19 05:02:26, "Joseph Turian" ***@***.***> wrote:

@SEMLLYCAT You can load htdemucs checkpoints into this repo. https://github.com/facebookresearch/demucs

**Comment by SEMLLYCAT** [comment]
In addition, I tried to use the official pre-training model to load into demucs_lightning, but encountered many problems
















At 2023-02-19 05:02:26, "Joseph Turian" ***@***.***> wrote:

@SEMLLYCAT You can load htdemucs checkpoints into this repo. https://github.com/facebookresearch/demucs

**Comment by SEMLLYCAT** [comment]
I don't know how to load the official pre-training model into the demucs_lightning version, please solve this problem at your convenience，thank you！
















At 2023-02-19 05:02:26, "Joseph Turian" ***@***.***> wrote:

@SEMLLYCAT You can load htdemucs checkpoints into this repo. https://github.com/facebookresearch/demucs
